### PR TITLE
Make omnibus/Gemfile support environment variables

### PR DESCRIPTION
**Pull request by tduffield**
## Description

This will allow us to dynamically define where we pull `omnibus` and `omnibus-software` in our adhoc builds. This will allow us to more easily test changes to those libraries without having to make branches on this repository. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: omnibus/Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source "https://rubygems.org"
 
-gem "omnibus", git: "https://github.com/chef/omnibus.git", branch: "master"
-gem "omnibus-software", git: "https://github.com/chef/omnibus-software.git", branch: "master"
+gem 'omnibus', github: ENV.fetch('OMNIBUS_GITHUB_REPO', 'chef/omnibus'), branch: ENV.fetch('OMNIBUS_GITHUB_BRANCH', 'master')
```
Comment (tas50) [inline omnibus/Gemfile:3]: We'll need to confirm if this breaks dependabot. They tend to just fail if you put any logic into your gemfile for security reasons.

Comment (tduffield) [reply omnibus/Gemfile:3]: It doesn't appear to, at least with the native GitHub-based version: https://github.com/chef/omnibus-harmony/network/dependencies

Comment (tduffield) [reply omnibus/Gemfile:3]: The logs from the scan itself seem to indicate that dependabot has no trouble with this particular logic.

Comment (tas50) [reply omnibus/Gemfile:3]: Nice

**Pull request merged**